### ISSUE TITLE
Cannot derive `Send` bound when a value of associated type with lifetime bound where clause across `await` point.

### ISSUE THREAD

**Issue by zirconium-n** [rust-lang/rust#93323]
I tried this code:

```rust
#![feature(generic_associated_types)]

trait Foo<'ctx, 'env> {
    type X: Send + Sync + 'env
    where
        'env: 'ctx;
}

fn foo<'ctx, 'env: 'ctx, T: Foo<'ctx, 'env>>(x: T::X) -> impl Send + 'env + 'ctx {
    async move {
        let x = x;
        std::future::ready(()).await;
    }
}
```

I expected to see this happen: compile success

Instead, this happened: compile failed with error message:
```
error[E0478]: lifetime bound not satisfied
 --> src/main.rs:9:58
   |
9 | fn foo<'ctx, 'env: 'ctx, T: Foo<'ctx, 'env>>(x: T::X) -> impl Send + 'env + 'ctx {
   |   
```
The error message says it's a lifetime bound but it's not. If I change `impl Send` to `impl Sized` then it compiles.

### Meta

`rustc --version --verbose`:
```
rustc 1.60.0-nightly (51126be1b 2022-01-24)
binary: rustc
commit-hash: 51126be1b260216b41143469086e6e6ee567647e
commit-date: 2022-01-24
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 13.0.0
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by zirconium-n** [comment]
The code itself does not use GAT, but where bound on associated types is feature-gated by GAT.

**Comment by zirconium-n** [comment]
Seems a duplicate of https://github.com/rust-lang/rust/issues/64552,
